Linux/sgx/test_app/enclave/tests/bntest.c

```diff
@@ -342,12 +342,14 @@ int bn_test()
 
     return 0;
  err:
+    BN_CTX_free(ctx);
     BIO_puts(out, "1\n");       /* make sure the Perl script fed by bc
                                  * notices the failure, see test_bn in
                                  * test/Makefile.ssl */
     (void)BIO_flush(out);
 
     ERR_print_errors_fp(stderr);
+    BIO_free(out);
```

Comment:
It looks better to initialize out to NULL at the beginning of this function.

Reply:
I'll initialize to NULL at the beginning of the function as you suggested but we need to call BIO_free to release the resource before the function returns the same way we do for the successful path in line 341.